### PR TITLE
Split out types of erc721 and erc1155 and erc20 with more

### PR DESCRIPTION
**Pull request by ponyjackal**

### CODE REVIEW

File: src/types/asset.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+export enum AssetType {
+  ETH = 'eth',
+  ERC20 = 'erc20',
+  ERC721 = 'erc721',
+  ERC1155 = 'erc1155',
+}
+
+export type ETHAsset = {
+  id: string;
+  asset: string;
```
Comment (pcowgill) [inline src/types/asset.ts:10]: @ponyjackal What's an example value for this string?

Comment (ponyjackal) [reply src/types/asset.ts:10]:
```
export const WETH_ADDRESSES = [
  '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2', // Ethereum
  '0x4200000000000000000000000000000000000006', // OP Stack
  '0xe5d7c2a44ffddf6b295a15c148167daaaf5cf34f', // Linea
  '0x0000000000a39bb272e79075ade125fd351887ac', // Blur
];
```

Comment (ponyjackal) [reply src/types/asset.ts:10]: thats a good catch, I think we wont need asset string here

File: src/types/context.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  unit?: string; // nonce is an example where there is no unit
+};
+
+export type ContextLinkType = {
+  type: 'link' | 'schemaID';
+  value: string;
+  link: string;
+};
+
+export type ContextSummaryVariableType =
```
Comment (pcowgill) [inline src/types/context.ts:85]: @ponyjackal This is much cleaner now, nice!

File: src/types/context.ts [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+  type: AssetType.ERC721;
+  tokenId: string;
+  token: string;
+  indexed?: boolean;
+  emphasis?: boolean;
+};
+
+export type ContextERC1155Type = {
+  type: AssetType.ERC1155;
+  tokenId: string;
```
Comment (pcowgill) [inline src/types/context.ts:49]:
```suggestion
```

**Comment by pcowgill** [comment]
@ponyjackal Can you please take a look at the build failure? Thanks!

**Comment by ponyjackal** [comment]
> @ponyjackal Can you please take a look at the build failure? Thanks!

@pcowgill This PR is not finished yet for tokenMint, which I will finish up once [This PR](https://github.com/Once-Upon/context/pull/136) is merged

**Comment by pcowgill** [comment]
> > @ponyjackal Can you please take a look at the build failure? Thanks!
> 
> @pcowgill This PR is not finished yet for tokenMint, which I will finish up once [This PR](https://github.com/Once-Upon/context/pull/136) is merged

@ponyjackal That PR is merged now.

**Review by pcowgill — APPROVED** [review]

**Pull request merged**